src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractPackage`2.cs

```diff
@@ -76,6 +75,12 @@ protected override void Initialize()
                 this.Workspace.StartSolutionCrawler();
             }
 
+            // start remote host if enabled
+            if (this.Workspace.Options.GetOption(InternalFeatureOnOffOptions.RemoteHost))
```

Comment:
I'd prefer the option move into the remote host implementation.  That's what we do with other features (like nuget search).  That way the package just says "start/stop" and the actual service says "actually, i'm disabled, i'm not going to do anything".